### PR TITLE
add Auth0 configuration documentation

### PR DESCRIPTION
**Pull request by philvarner**

### CODE REVIEW

File: doc/source/authentication.rst [hunk 1]
```diff
@@ -208,6 +208,28 @@ and obtain the confidential client credentials.
          userdata_params: {'state': 'state'}
          username_key: preferred_username
 
+Auth0
+^^^^^^^^^^^^^^
```
Comment (willingc) [inline doc/source/authentication.rst]: Thanks for the PR @philvarner. A few comments...

Comment (willingc) [reply doc/source/authentication.rst]: In rst, the underline should be the same length as the line above.

File: doc/source/authentication.rst [hunk 2]
```diff
@@ -208,6 +208,28 @@ and obtain the confidential client credentials.
...
+
+Auth0 is a popular commercial provider of identity management. The jupyterhub helm does not support Auth0 as a
```
Comment (willingc) [inline doc/source/authentication.rst]: ...JupyterHub helm chart does not include Auth0 support by default. To use Auth0, ``extraEnv`` and ``extraConfig`` need to be configured as follows:

File: doc/source/authentication.rst (hunk 1 above)
Comment (philvarner) [reply doc/source/authentication.rst]: > In rst, the underline should be the same length as the line above.

fixed

File: doc/source/authentication.rst (hunk 2 above)
Comment (philvarner) [reply doc/source/authentication.rst]: Thanks, I merged what we both wrote.

File: doc/source/authentication.rst [hunk 3]
```diff
@@ -208,6 +208,31 @@ and obtain the confidential client credentials.
...
+Auth0 is a popular commercial provider of identity management. The JupyterHub helm chart does not include support for
+Auth0 by default.  To use Auth0, ``extraEnv`` and ``extraConfig`` must be configured as follows:
+
+Note that without the scope defined, authenticating to JupyterHub after already being logged in to Auth0 will fail.
+
+.. code-block:: yaml
+
+    hub:
+      extraEnv:
+        AUTH0_SUBDOMAIN: 'prod-8ua-1yy9'
```
Comment (davidsmf) [inline doc/source/authentication.rst:223]: It looks like `extraEnv` needs to be in `name:`, `value:` format.

```suggestion
      - name: AUTH0_SUBDOMAIN
        value: 'prod-8ua-1yy9'
```

Comment (consideRatio) [reply doc/source/authentication.rst:223]: This Helm chart supports the current format alongside the name/value format with is the k8s native format.

File: doc/source/authentication.rst [hunk 4]
```diff
@@ -208,6 +208,31 @@ and obtain the confidential client credentials.
...
+          c.JupyterHub.authenticator_class = 'oauthenticator.auth0.Auth0OAuthenticator'
+          c.Auth0OAuthenticator.client_id = 'y0urc1logonc1ient1d'
+          c.Auth0OAuthenticator.client_secret = 'an0ther1ongs3cretstr1ng'
+          c.Auth0OAuthenticator.oauth_callback_url = 'https://<your_jupyterhub_host>/hub/oauth_callback'
+          c.Auth0OAuthenticator.scope = ['openid', 'email']
+          c.Authenticator.admin_users = {
+                'devops@example.com'
+              }
+          c.Authenticator.auto_login = True
+
```
Comment (consideRatio) [inline doc/source/authentication.rst:235]:
```suggestion
          c.JupyterHub.authenticator_class = 'oauthenticator.auth0.Auth0OAuthenticator'
          c.Auth0OAuthenticator.client_id = 'client-id-from-auth0-here'
          c.Auth0OAuthenticator.client_secret = 'client-secret-from-auth0-here'
          c.Auth0OAuthenticator.oauth_callback_url = 'https://<your_jupyterhub_host>/hub/oauth_callback'
          c.Auth0OAuthenticator.scope = ['openid', 'email']
          c.Authenticator.admin_users = {
              'devops@example.com'
          }
          c.Authenticator.auto_login = True
```

**Comment by consideRatio** [comment]
Thanks @philvarner, sorry for the extremely slow turnaround on getting this PR merged =/

**Pull request merged**